pillarbox-demo-tv/src/main/java/ch/srgssr/pillarbox/demo/tv/ui/integrationLayer/ListsHome.kt

```diff
@@ -0,0 +1,329 @@
+import androidx.tv.material3.MaterialTheme
+import androidx.tv.material3.Text
+import ch.srgssr.pillarbox.demo.shared.ui.NavigationRoutes
+import ch.srgssr.pillarbox.demo.shared.ui.integrationLayer.ContentList
+import ch.srgssr.pillarbox.demo.shared.ui.integrationLayer.data.ContentListSection
+import ch.srgssr.pillarbox.demo.shared.ui.integrationLayer.data.contentListSections
+import ch.srgssr.pillarbox.demo.tv.ui.theme.PillarboxTheme
+
+/**
+ * Screen of the "Lists" tab of the demo app on TV.
+ *
+ * @param sections The list of section to display.
+ * @param modifier The [Modifier] to apply to this screen.
+ *
+ * @see ContentListSection
+ */
+@Composable
+@OptIn(ExperimentalTvMaterial3Api::class)
+fun ListsHome(
+    sections: List<ContentListSection>,
+    modifier: Modifier = Modifier
+) {
+    val navController = rememberNavController()
+
+    NavHost(
+        navController = navController,
+        startDestination = NavigationRoutes.contentLists,
+        modifier = modifier.fillMaxSize()
+    ) {
+        composable(NavigationRoutes.contentLists) {
+            ListsSection(
+                items = sections,
+                itemToString = { it.title },
+                onItemClick = { index, _ ->
+                    navController.navigate("${NavigationRoutes.contentList}/$index")
+                }
+            )
+        }
+
+        composable(
+            route = "${NavigationRoutes.contentList}/{index}",
+            arguments = listOf(
+                navArgument("index") { type = NavType.IntType }
+            )
+        ) {
+            val sectionIndex = it.arguments?.getInt("index") ?: 0
+            val section = sections[sectionIndex]
+
+            BackHandler {
+                navController.popBackStack()
+            }
+
+            ListsSection(
+                title = section.title,
+                items = section.contentList,
+                itemToString = { item ->
+                    when (item) {
+                        is ContentList.ContentListWithBu -> item.bu.name.uppercase()
+                        is ContentList.RadioLatestMedias -> item.radioChannel.label
+                        is ContentList.RadioShows -> item.radioChannel.label
+                        else -> ""
+                    }
+                },
+                onItemClick = { _, contentList ->
+                    navController.navigate(contentList.getDestinationRoute())
+                }
+            )
+        }
+
+        composable(
+            route = ContentList.TvTopics.route,
+            arguments = listOf(
+                navArgument("bu") { type = NavType.StringType }
+            )
+        ) {
+            val contentList = ContentList.TvTopics.parse(it)
+
+            BackHandler {
+                navController.popBackStack()
+            }
+
+            // TODO Integrate content (https://github.com/SRGSSR/pillarbox-android/issues/298)
+            Text(text = contentList.toString())
+        }
```

Comment:
I notice that this code is used multiple times. Can we factorize it in some way?

Reply:
I hope so. For now, I've just reused what @StaehliJ created for the mobile demo app.
I'll investigate this more while working on https://github.com/SRGSSR/pillarbox-android/issues/298, as I'll address these TODOs there 🙂